src/fal/utils.py

```diff
@@ -25,3 +26,25 @@ def next(self) -> int:
     def __int__(self) -> int:
         """Return the last value."""
         return self._index
+
+
+class _ReInitialize:
+    def __init__(self, *args, **kwargs):
+        self._serialization_state = {
+            "args": copy.deepcopy(args),
+            "kwargs": copy.deepcopy(kwargs),
+        }
+        super().__init__(*args, **kwargs)
+
+    def __getstate__(self):
+        return self._serialization_state
+
+    def __setstate__(self, state):
+        super().__init__(*state["args"], **state["kwargs"])
+
+
+def has_side_effects(cls):
```

Comment:
if we are just using this internally and the only purpose of this is to decide on a serilization strategy, should the name also reflect that? `reinitilize` might be a better name maybe?